crates/agent/src/api/snapshot.rs

```diff
@@ -13,6 +14,8 @@ pub struct Snapshot {
     pub collections_idx_name: Vec<usize>,
     // Platform data-planes.
     pub data_planes: tables::DataPlanes,
+    // Signing HMAC key of data-planes, keyed on data_plane_name
```

Comment:
Can we not continue to store these in `tables::DataPlane.hmac_keys` after decryption?

`tables::DataPlanes` implements Clone as well, so you can clone and pass into successive try_fetch just as you do with the HashMap

A nice-to-have would be to wrap tables::DataPlane.hmac_keys in zeroize as well. Not a deal breaker if that's non-trivial. I don't recall why i didn't do that initially, there may have been a challenge

Reply:
@jgraettinger if we store them in `hmac_keys`, after we are done with migration and want to remove unencrypted `hmac_keys` from the database table, we will have a discrepency between the `tables::DataPlane` model and the actual table, which I would rather avoid. I think `sqlx` would complain as well.

Basically, it would work for this transition step, but afterwards we would be changing it again 🤷🏽